### PR TITLE
task(settings): Update grey and blue links' hover and focus states

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We want to improve the the behavior on links when they are hovered over and focused on.

## This pull request

- Moves colors-grey:600 to colors-grey:900
- Introduces new grey:600 that is in line with figma
- Introduces new gray:700 that is in line with figma
- Updates all occurrences of text-grey-600 to text-grey-900 so existing styling doesn't change
- Applies focus and focus-visible states to link-grey and link-blue as per figma
- Updates link-grey:hover state to be inline with figma
- Updates link-grey:active and link-blue:active to be inline with figma

## Issue that this pull request solves

Closes: #13865

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### Grey Links
normal
image

:hover
image

:active
image

:focus
image

:focus-visible
image

### Blue Links

normal
image

:hover
image

:active
image

:focus
image

:focus-visible
image

### Figma spec
image

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

**Comment by pdehaan** [comment]
Yikes, not sure if this is correct, but this janky <kbd>grep</kbd> seems to find as many [unique] `text-grey-*` strings as I can find.

```sh
git grep -EPoh "[\s\b\"](\w+:)?text-grey-\d+\b" | sort | uniq

 active:text-grey-400
 active:text-grey-700
 active:text-grey-900
 disabled:text-grey-300
 focus:text-grey-900
 hover:text-grey-400
 hover:text-grey-500
 hover:text-grey-900
 text-grey-300
 text-grey-400
 text-grey-50
 text-grey-500
 text-grey-600
 text-grey-900
"text-grey-100
"text-grey-400
"text-grey-500
"text-grey-900
```

**Comment by dschom** [comment]
> Yikes, not sure if this is correct, but this janky grep seems to find as many [unique] `text-grey-*` strings as I can find.

@pdehaan `text-grey-*` is used in many scenarios (links, text, buttons, etc.), so I don't take issue with this. All of those text-grey values are defined by the `fax-react/config/tailwind.js` config file and are valid tailwind text classes.

It's important to note that in tailwind a `hover:`  prefix isn't a unique class name but just short hand. Essentially these two declarations do the same thing:

```
.some-class { 
@apply text-grey-500 hover:text-gray-900;
}
```

vs.

```
.some-class {
  @apply text-grey-500;  
}
.some-class:hover {
 @apply text-grey-900;
}
```

So as far a uniqueness goes, a simpler search, `git grep -EPoh "text-grey-\d+\b" | sort | uniq`, gives us a more reasonable set of unique values:

```
text-grey-100
text-grey-300
text-grey-400
text-grey-50
text-grey-500
text-grey-600
text-grey-700
text-grey-900
```

File: packages/fxa-content-server/app/styles/_layout.scss [hunk 1]
```diff
@@ -91,7 +91,7 @@
     cursor: pointer; // Use the correct cursor for anchors without an href
     text-decoration: none;
 
-    @include anchor-default-pseudo-classes();
+    // @include anchor-default-pseudo-classes();
```
Comment (LZoog) [inline packages/fxa-content-server/app/styles/_layout.scss:94]: I think we need this though maybe you meant to uncomment it before pushing. There's a comment further up that mentions this should only affect pages that aren't using TW / our TW component classes yet (they've got an ID of `main-content`). If you check out `/reset_password` for example, the "Remember password? Sign in" link doesn't have hover/active styles.

File: packages/fxa-settings/src/components/PageAvatar/buttons.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { ReactComponent as ZoomInIcon } from './zoom-in.svg';
 import { ReactComponent as RotateIcon } from './rotate.svg';
 
 const buttonClass = `mx-2 text-grey-500
-hover:text-grey-600 hover:text-grey-600
+hover:text-grey-900 hover:text-grey-900
```
Comment (LZoog) [inline packages/fxa-settings/src/components/PageAvatar/buttons.tsx:21]: Not really part of your PR but `hover:text-grey-900` is on here twice. 🤔

**Review by LZoog — APPROVED** [review]
Looks great, thank you and thanks for the screenshots! 😁

I figure you meant to uncomment the commented out line in `_layout.scss` so I didn't block r+ on it - I can circle back if there's misunderstanding there.

File: packages/fxa-content-server/app/styles/_layout.scss (hunk 1 above)
Comment (dschom) [reply packages/fxa-content-server/app/styles/_layout.scss:94]: Yes, this was just me experimenting. I am glad you spotted this! Thank you.

File: packages/fxa-settings/src/components/PageAvatar/buttons.tsx (hunk 2 above)
Comment (dschom) [reply packages/fxa-settings/src/components/PageAvatar/buttons.tsx:21]: Interesting, I will clean this up anyway.

**Pull request merged**